.github/workflows/release.yml

```diff
@@ -1,42 +1,46 @@
-name: "release"
+name: Release
+
 on:
   push:
-    tags:
-      - "v*"
+    branches:
+      - main
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
 
 jobs:
-  publish:
-    runs-on: "ubuntu-22.04"
-    permissions:
-      id-token: write
-      contents: write
-      packages: write
-      pull-requests: write
-      issues: read
+  release:
+    name: Release
+    runs-on: ubuntu-latest
     steps:
-      - uses: "actions/checkout@v3"
-        with:
-          fetch-depth: 0
-      - uses: "actions/setup-node@v3"
-        with:
-          node-version: 18
-          cache: "npm"
-      - name: "Cancel previous runs"
-        uses: "styfle/cancel-workflow-action@0.11.0"
-        continue-on-error: true
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+
+      - name: Setup Node.js 18.x
+        uses: actions/setup-node@v3
         with:
-          access_token: ${{ secrets.GITHUB_TOKEN }}
-      - run: "npm i"
-      - run: "npm run build"
-      - name: "release / pull_request"
-        id: "changesets"
-        uses: "dotansimha/changesets-action@v1.5.2"
+          node-version: 18.x
+          registry-url: "https://registry.npmjs.org"
+
+      - name: Install Dependencies
+        run: npm install
+
+      - name: Build
+        run: npm run build
+
+      - name: Create Release Pull Request or Publish to npm
+        id: changesets
+        uses: changesets/action@v1
         with:
-          publish: "npm run release"
-          version: "npm exec changeset version"
-          commit: "Update stagehand version"
-          title: "Upcoming Release Changes"
-          createGithubReleases: "aggregate"
+          publish: npm run release
```

Comment:
i think you need to build the package before releasing it right? i'll reread the action, but pretty sure this will result in publishing an empty tarball

Reply:
looks like publish is supposed to build and publish: https://github.com/changesets/action/blob/31ff97e8acd3cfd08714eab3721f55da10003255/action.yml#L7-L8

Reply:
an example from changesets as well: https://github.com/changesets/changesets/blob/f03f70bffc9a033747f45b0d071c6c2ed8818ac7/.github/workflows/changeset-version.yml#L25-L26

Reply:
I might've committed this after you commented, but we do run `npm run build` right before this

Reply:
I _think_ this should be addressed